src/xmipp/libraries/data/monogenic.h

```diff
@@ -0,0 +1,172 @@
+
+
+	//FOURIERFREQS_3D: Determine the map of frequencies in the Fourier Space as an output.
+	// Also, the accesible frequencies along each direction are calculated, they are
+	// "freq_fourier_x", "freq_fourier_y", and "freq_fourier_z".
+	MultidimArray<double> fourierFreqs_3D(const MultidimArray< std::complex<double> > &myfftV,
+			const MultidimArray<double> &inputVol,
+			Matrix1D<double> &freq_fourier_x,
+			Matrix1D<double> &freq_fourier_y,
+			Matrix1D<double> &freq_fourier_z);
+
+	//RESOLUTION2EVAL: Determines the resoltion to be analzed in the estimation
+	//of the local resolution. These resolution are freq, freqL (diginal units)
+	// being freqL the tail of the raise cosine centered at freq. The parameter
+	// resolution is the frequency freq in converted into Angstrom. The parameters
+	//minRes and maxRes, determines the limits of the resolution range to be
+	// analyzed (in Angstrom). Sampling is the sampling rate in A/px, and step,
+	// is the resolution step for which analyze the local resolution.
+	void resolution2eval(int &count_res, double step,
+			double &resolution, double &last_resolution,
+			double &freq, double &freqL,
+			int &last_fourier_idx,
+			int &volsize,
+			bool &continueIter,	bool &breakIter,
+			double &sampling, double &minRes, double &maxRes,
+			bool &doNextIteration);
+
+	// AMPLITUDEMONOSIG3D_LPF: Estimates the monogenic amplitude of a HPF map, myfftV is the
+	// Fourier transform of that map. The monogenic amplitude is "amplitude", and the
+	// parameters "fftVRiesz", "fftVRiesz_aux", "VRiesz", are auxiliar variables that will
+	// be defined inside the function (they are input to speed up monores algorithm).
+	// Freq, freqL and freqH, are the frequency of the HPF and the tails of a raise cosine.
+	// Note that once the monogenic amplitude is estimated, it is low pass filtered to obtain
+	// a smooth version and avoid rippling. freq_fourier_x, freq_fourier_y, freq_fourier_z,
+	// are the accesible Fourier frequencies along each axis.
+	void amplitudeMonoSig3D_LPF(const MultidimArray< std::complex<double> > &myfftV,
+			FourierTransformer &transformer_inv,
+			MultidimArray< std::complex<double> > &fftVRiesz,
+			MultidimArray< std::complex<double> > &fftVRiesz_aux, MultidimArray<double> &VRiesz,
+			double freq, double freqH, double freqL, MultidimArray<double> &iu,
+			Matrix1D<double> &freq_fourier_x, Matrix1D<double> &freq_fourier_y,
+			Matrix1D<double> &freq_fourier_z, MultidimArray<double> &amplitude,
+			int count, FileName fnDebug);
+
+	// STATISTICSINBINARYMASK2: Estimates the staticstics of two maps:
+	// Signal map "volS" and Noise map "volN". The signal statistics
+	// are obtained by mean of a binary mask. The results are the mean
+	// and variance of noise and signal, as well as the number of voxels
+	// of signal and noise NS and NN respectively. The thr95 represents
+	// the percentile 95 of noise distribution.
```

Comment:
preferably return some struct, instead of passing so many params.

Reply:
you are right, it is a major change,  I will think about it in next pull request of this class